src/main/java/ToDoApplication/DAO/ToDoDAO.java

```diff
@@ -32,6 +34,21 @@ public String addToDo(ToDo todo){
         } catch (SQLException e) {
             e.printStackTrace();
             msg = "failure adding new entry";
+        } finally {
+            if (connection != null){
+                try {
```

Comment:
there are a lot of repetitive code. What do you think of making abstract class (e.g. BaseDao) and make it parent class for ToDoDao and UserDao. In BaseDao class make protected method closeResources(Connection con, PreparedStatement pstmt, ResultSet rs)?
And put the code from finally there. And in all finally blocks just call the method closeResources(con, pstmt, rs).